### PR TITLE
Split out managed resources into separate classes

### PR DESCRIPTION
**Pull request by ananace**
Having a class structure for K8s resources would make it more obvious where to put other Puppet-managed resources when expanding the module, and it should also make it easier to deploy only loose resources.

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -7,30 +7,31 @@
...
-* [`k8s::repo`](#k8s--repo): Handles repositories for the container runtime
-* [`k8s::server`](#k8s--server): Sets up a Kubernetes server instance
-* [`k8s::server::apiserver`](#k8s--server--apiserver): Installs and configures a Kubernetes apiserver
-* [`k8s::server::controller_manager`](#k8s--server--controller_manager): Installs and configures a Kubernetes controller manager
-* [`k8s::server::etcd`](#k8s--server--etcd): Sets up an etcd cluster node
-* [`k8s::server::etcd::setup`](#k8s--server--etcd--setup): Installs and configures an etcd instance
-* [`k8s::server::resources`](#k8s--server--resources): Generates and deploys standard Kubernetes in-cluster services
-* [`k8s::server::scheduler`](#k8s--server--scheduler): Installs and configures a Kubernetes scheduler
-* [`k8s::server::tls`](#k8s--server--tls): Generates the necessary Kubernetes certificates for a server
+* [`k8s::install::kubeadm`](#k8sinstallkubeadm): Installs the kubeadm binary
```
Comment (bastelfreak) [inline REFERENCE.md:10]: looks like you generated it with puppet-strings 2. Please update your puppet-strings to version 3 (that's the version also used in the CI).

Comment (ananace) [reply REFERENCE.md:10]: Interesting, this was done from a fresh clone and bundle install, so apparently there's a `puppet-strings` dependency that's too lenient.

**Comment by rwaffen** [comment]
i like this approach, the resources class is far to big atm (for me). 👍

File: manifests/server/resources/kube_proxy.pp [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+          },
+        ],
+        roleRef  => {
+          kind     => 'ClusterRole',
+          name     => 'system:node-proxier',
+          apiGroup => 'rbac.authorization.k8s.io',
+        },
+      };
+
+    # apiVersion: kubeproxy.config.k8s.io/v1alpha1
```
Comment (rwaffen) [inline manifests/server/resources/kube_proxy.pp:68]: why this comment block, what does it say? looks like a yaml block.

**Review by rwaffen — APPROVED** [review]
LGTM

File: manifests/server/resources/kube_proxy.pp (hunk 2 above)
Comment (ananace) [reply manifests/server/resources/kube_proxy.pp:71]: Threw a comment on it, to make it look less arbitrary.

**Pull request merged**